### PR TITLE
make graph series error bars enable/disable with their matched series

### LINKED ISSUE #9310: Error bars still visible after series are disabled
### What happened?

In case it is not intentional. 

image

### What Operating System are you on?

Windows

### PR DESCRIPTION
**Pull request by ric394**
resolves #9310

# Notes
- Error bars for a series are actually a separate series. Special handling is required to have them toggle visibility inline with their matched scatter plot.

### CODE REVIEW

**Comment by hol353** [comment]
retest this please jenkins

File: APSIM.Shared/APSIM.Shared.csproj [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <PackageReference Include="SkiaSharp.NativeAssets.Linux" Version="2.88.6" />
     <PackageReference Include="System.Drawing.Common" Version="6.0.0" />
     <PackageReference Include="Docker.DotNet" Version="3.125.13" />
+    <PackageReference Include="OxyPlot.Core" Version="2.1.2" />
```
Comment (hol353) [inline APSIM.Shared/APSIM.Shared.csproj:27]: Do we need to add this dependency to APSIM.Shared?

File: APSIM.Shared/Graphing/NamedScatterErrorSeries.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using OxyPlot.Series;
+
+namespace APSIM.Shared.Graphing;
+
+/// <summary>
+/// A Nameable ScatterErrorSeries.
+/// </summary>
+public class NamedScatterErrorSeries: ScatterErrorSeries, INameableSeries
```
Comment (hol353) [inline APSIM.Shared/Graphing/NamedScatterErrorSeries.cs:8]: Hmm this doesn't seem to do much. Why is it necessary?

If it is needed then move it to where it is needed (not in APSIM.Shared). One day I'd like to remove all the graphing code out of APSIM.Shared.

File: APSIM.Shared/APSIM.Shared.csproj (hunk 1 above)
Comment (ric394) [reply APSIM.Shared/APSIM.Shared.csproj:27]: In it's current implementation it's required but I may be able to move it to ApsimNG

File: APSIM.Shared/Graphing/NamedScatterErrorSeries.cs (hunk 2 above)
Comment (ric394) [reply APSIM.Shared/Graphing/NamedScatterErrorSeries.cs:8]: @hol353 Alternatively, would you suggest making a separate assembly for graphing in this pull request to start the process of moving graphing related files to their own assembly?

Comment (hol353) [reply APSIM.Shared/Graphing/NamedScatterErrorSeries.cs:8]: No, don't create a new assembly. That's a bigger job for another day.

**Pull request merged**